mail_activity_reminder/models/mail_activity.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2020 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import datetime, time
+from dateutil.relativedelta import relativedelta
+from pytz import timezone, UTC
+
+from odoo import _, api, fields, models
+
+
+class MailActivity(models.Model):
+    _inherit = 'mail.activity'
+
+    next_reminder = fields.Datetime(
+        string='Next reminder',
+        compute='_compute_next_reminder',
+        compute_sudo=True,
+        store=True,
+    )
+    last_reminder_local = fields.Datetime(
+        string='Last reminder (local)',
+    )
+
+    @api.model
+    def _get_activities_to_remind_domain(self):
+        """Hook for extensions"""
+        return [('next_reminder', '<=', fields.Datetime.now())]
+
+    @api.model
+    def _get_activities_to_remind(self):
+        return self \
+            .search(self._get_activities_to_remind_domain()) \
+            .filtered(lambda activity: activity._should_be_reminded())
+
+    @api.model
+    def _process_reminders(self):
+        activities = self._get_activities_to_remind()
+        activities.action_remind()
+        return activities
+
+    @api.multi
+    def _should_be_reminded(self):
+        """Hook for extensions"""
+        self.ensure_one()
+        if self.next_reminder > fields.Datetime.now():
```

Comment:
this IF is already taken into account in the domain now

Reply:
That domain is optimization, so for clarity this method should perform a full check, not assuming "oh everything else is checked elsewhere"

Reply:
fine